pkg/dynamiclifecycle/manager.go

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamiclifecycle
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/rate"
+	"github.com/cilium/cilium/pkg/time"
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/reconciler"
+
+	"github.com/cilium/cilium/pkg/dynamicconfig"
+)
+
+type FeatureStatus struct {
+	Feature DynamicFeature
+	Enabled bool
+}
+
+type Manager interface {
```

Comment:
Do we need the interface for this? Could just drop this and have `type Manager struct`.